src/coins.cpp

```diff
@@ -1298,18 +1339,19 @@ bool CCoinsViewCache::HandleSidechainEvents(int height, CBlockUndo& blockUndo, s
                 __func__, __LINE__, maturingScId.ToString(), height);
 
         assert(HaveSidechain(maturingScId));
-        assert(cacheSidechains.at(maturingScId).scInfo.mImmatureAmounts.count(height));
+        CSidechainsMap::iterator scMaturingIt = ModifySidechain(maturingScId);
+        assert(scMaturingIt->second.scInfo.mImmatureAmounts.count(height));
 
-        cacheSidechains.at(maturingScId).scInfo.balance += cacheSidechains.at(maturingScId).scInfo.mImmatureAmounts[height];
+        scMaturingIt->second.scInfo.balance += AccessSidechain(maturingScId)->mImmatureAmounts.at(height);
```

Comment:
AccessSidechain is not needed, use scMaturingIt->second.scInfo.mImmatureAmounts

Reply:
Done